player/configfiles.c

```diff
@@ -396,27 +396,27 @@ void mp_delete_watch_later_conf(struct MPContext *mpctx, const char *file)
     talloc_free(fname);
 }
 
-void mp_load_playback_resume(struct MPContext *mpctx, const char *file)
+bool mp_load_playback_resume(struct MPContext *mpctx, const char *file)
 {
     if (!mpctx->opts->position_resume)
-        return;
+        return false;
     char *fname = mp_get_playback_resume_config_filename(mpctx, file);
     if (fname && mp_path_exists(fname)) {
         if (mpctx->opts->position_check_mtime &&
             !mp_is_url(bstr0(file)) && !check_mtime(file, fname))
         {
             talloc_free(fname);
-            return;
+            return false;
         }
 
         // Never apply the saved start position to following files
         m_config_backup_opt(mpctx->mconfig, "start");
         MP_INFO(mpctx, "Resuming playback. This behavior can "
                "be disabled with --no-resume-playback.\n");
         try_load_config(mpctx, fname, M_SETOPT_PRESERVE_CMDLINE, MSGL_V);
-        unlink(fname);
     }
     talloc_free(fname);
+    return true;
```

Comment:
This is kinda nitpicky and technically this works fine as is, but returning true from this function is supposed to mean that it did resume playback right? But if fname is NULL or the file doesn't exist, playback also doesn't get resumed and so returning true isn't right.
```suggestion
        talloc_free(fname);
        return true;
    }
    return false;
```

Reply:
No that's totally fair and an oversight on my part. `talloc_free` always needs to be called though so I did it a literally differently.